synth_node_bin/src/main.rs

```diff
@@ -102,24 +102,20 @@ async fn run_synth_node(node_addr: SocketAddr) -> Result<()> {
 //
 // All the program logic happens here.
 #[allow(unused_variables)]
-async fn perform_action(synth_node: &mut SyntheticNode) {
+async fn perform_action(synth_node: &mut SyntheticNode, addr: SocketAddr) {
```

Comment:
Just thinking if `perform_action` should return `Result<()>` as `run_synth_node` returns that. 
Now we've a path, where if `unicast` fails, there will be warning message displayed, `run_synth_node` will go further and return `Ok(())` to the `main` instead of `Err`. So we'll be returning `Ok(())` even if there was something wrong.

For now it doesn't matter but maybe at some point in the future we'll use it more extensively.

Reply:
Done!
For now, our binary shouldn't panic easily, let's monitor it!